### PR TITLE
Moved pflua-quickcheck into the main pflua repository

### PR DESCRIPTION
**Pull request by kbara**
It's not integrated into the main Makefile yet; it's a minimal set of changes to make the tests run against current master.

### CODE REVIEW

File: src/pf/backend.lua [hunk 1]
```diff
@@ -353,17 +353,30 @@ function emit_and_load(ssa, name)
    return func()
 end
 
+local function ast_to_ssa(ast)
+   local convert_anf = require('pf.anf').convert_anf
```
Comment (andywingo) [inline src/pf/backend.lua]: I think this is a layering violation; probably don't want this here.  Given that pflua-quickcheck will be building internal AST data structures (and possibly SSA / ANF structures too) it makes more sense to me to have this in the quickcheck module.

**Comment by andywingo** [comment]
What do you think the ultimate directory structure should be?  I guess it never makes sense to install the quickcheck stuff, unless we can generate test packets without a trace, since we shouldn't install sample traces.  Also consider that we will import pflua into snabb at times, and probably we shouldn't import traces, and probably don't need to import the check code (though it can't hurt).  Dunno.

**Comment by kbara** [comment]
Probably src/, tools/, and tests/ would be a sane directory structure, with subdirectories of tests/ for different types of test.
I'd quite like to have the QuickCheck tests being run as part of CI, so that we'll catch (more of the) bugs introduced by small changes sooner; we've had things as simple as syntactic errors breaking important paths for weeks before. It could usefully run on traces of a single packet, which would add rather little to the repository size; generating packets probably ends up being larger, though is also a possibility. I agree that a large pcap file shouldn't be part of either pflua itself or Snabb, or of every build/check of pflua.
I see the QuickCheck stuff as being a very lightweight form of regression testing that takes less work to modify than dozens or hundreds of internal tests that need to be changed by hand every time pflua internals changed. It's small and useful.

**Comment by andywingo** [comment]
src/, tools/, and tests/ sounds great to me; and no argument about the utility of quickcheck :)  Let's get in a version without the backend.lua changes -- if there is need for a centralized interface to avoid duplication we can put it in a module in tests/.  Sound OK?  Then we fix the directory structure.

**Comment by lukego** [comment]
I wonder if we could put the quick check implementation to use in Snabb Switch in the future, too? @eugeneia and I have been talking along these lines with no particular tool in mind.

I have never used QuickCheck in anger but I love the concept. I even had discussions with Quviq about ideas for testing the NFV application in the beginning.

**Comment by andywingo** [comment]
lgtm

@lukego we were chatting about this and that makes sense; we're still figuring out what kind of interface  to expose, but once it's in pflua and pflua is in snabb we'll know more about what that would look like.

**Pull request merged**